src/vs/workbench/contrib/notebook/browser/viewModel/notebookViewModelImpl.ts

```diff
@@ -910,7 +910,20 @@ export class NotebookViewModel extends Disposable implements EditorFoldingStateD
 	//#region Find
 	find(value: string, options: INotebookSearchOptions): CellFindMatchWithIndex[] {
 		const matches: CellFindMatchWithIndex[] = [];
-		this._viewCells.forEach((cell, index) => {
+		let findCells: CellViewModel[] = [];
+
+		const selectedRanges = options.selectedRanges?.map(range => this.validateRange(range));
+		if (options.searchInRanges && selectedRanges) {
+			for (const range of selectedRanges.filter(range => !!range)) {
```

Comment:
we can use helper function `cellRangesToIndexes(selectedRanges).map(index => this._views[index])`